src/cast.c

```diff
@@ -2597,7 +2641,180 @@ bool isVoidArrayLiteral(Expression *e, Type *other)
         ((ArrayLiteralExp *)e)->elements->dim == 0);
 }
 
+/**
+ * This function should be called by iterateAliasThis.
+ * It is needed to collect of subtypes of the expression type.
+ * It checks if type hasn't been found and returns it if yes.
```

Comment:
Comment is fixed.
This function returns false anything, however it also returns expression through `*outexpr` parameter.
Function does it only if  type was found.